exe/arduino_ci.rb

```diff
@@ -520,83 +484,64 @@ def perform_example_compilation_tests(cpp_library, config)
 
     # having undefined platforms is a config error
     platforms.select { |p| ovr_config.platform_info[p].nil? }.each do |p|
-      assure("Platform '#{p}' is defined in configuration files") { false }
+      @log.assure("Platform '#{p}' is defined in configuration files") { false }
     end
 
     install_all_packages(platforms, ovr_config)
     install_arduino_library_dependencies(ovr_config.aux_libraries_for_build, "<compile/libraries>")
 
     platforms.each do |p|
       board = ovr_config.platform_info[p][:board] # assured to exist, above
-      attempt("Compiling #{example_name} for #{board}") do
-        ret = @backend.compile_sketch(example_path, board)
-        unless ret
-          puts "Last command: #{@backend.last_msg}"
-          puts @backend.last_err
+      compiled_ok = @log.attempt("Compiling #{example_name} for #{board}") do
+        @backend.compile_sketch(example_path, board)
+      end
+
+      # decode the JSON output of the compiler a little bit
+      unless compiled_ok
+        @log.inform_multiline("Compilation failure details") do
+          begin
+            # parse the JSON, and print out only the nonempty keys. indent them with 4 spaces in their own labelled sections
```

Comment:
The comment mentions added indentation, but I think it is missing from the code.

Reply:
`inform_multiline` implicitly indents, so the full magic there is `@log.inform_multiline(k) { @log.iputs(val) } unless val.strip.empty?`